static/app/utils/replays/hooks/useReplayList.tsx

```diff
@@ -42,8 +43,11 @@ function useReplayList({eventView, organization}: Options): Result {
   }, [api, organization, location, eventView]);
 
   useEffect(() => {
-    init();
-  }, [init]);
+    if (!querySearchRef.current || querySearchRef.current !== location.search) {
+      querySearchRef.current = location.search;
+      handleFetch();
+    }
+  }, [handleFetch, location.search]);
```

Comment:
i think there's a `usePrevious` which could be doing about the same thing

```
lastSearch = usePrevious(location.search)

if (lastSearch !== location.search) { ...
```

Reply:
I didn't use usePrevious here because it gets initialized to the first value right away, so it won't actually run because the search string doesn't change. The location.key and action type are what change causing the effects to re-fire. This way it runs initially and then only on actual changes to the query aftewards

Reply:
ahh, cool 👍